extensions/kube-docs/src/index.tsx

```diff
@@ -0,0 +1,49 @@
+import { Action, ActionPanel, Icon, List } from "@raycast/api";
+import { useCallback, useState } from "react";
+import { debounce } from "throttle-debounce";
+import { DocsItem, DocsResponse } from "./types";
+import useFetcher from "./hooks/useFetcher";
+
+const DEBOUNCE_DELAY = 300;
+const ICON = "command-icon.png";
+
+export default function main() {
+  const { loading, fetcher } = useFetcher();
+  const [items, setItems] = useState<DocsItem[]>([]);
+
+  const onQueryChange = async (query: string) => {
+    if (!query) {
+      setItems([]);
+      return;
+    }
+
+    const {
+      data: { webPages },
+    } = await fetcher<DocsResponse>({ query });
+
+    if (!webPages || !webPages.value) return;
+
+    setItems(webPages.value);
+  };
+
+  const debounceCallback = useCallback(debounce(DEBOUNCE_DELAY, onQueryChange), []);
```

Comment:
```suggestion
```
You shouldn't use a “custom” debounce, there is one handling this directly in Raycast, you can add `throttle` directly to the [<List> element](https://developers.raycast.com/api-reference/user-interface/list#list)